components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4569,4 +4569,32 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
+
+    public static String[] getRequestedOIDCScopes(String[] requestedScopes)
+            throws IdentityOAuthAdminException {
+
+        String[] oidcScopes =
+                OAuth2ServiceComponentHolder.getInstance().getOAuthAdminService().getScopeNames();
+        List<String> requestedOIDCScopes = new ArrayList<>();
+        for (String scope : requestedScopes) {
+            if (ArrayUtils.contains(oidcScopes, scope)) {
+                requestedOIDCScopes.add(scope);
+            }
+        }
+        return requestedOIDCScopes.toArray(new String[requestedOIDCScopes.size()]);
+    }
+
+    public static String[] removeOIDCScopesFromRequestedScopes(String[] requestedScopes)
+            throws IdentityOAuthAdminException {
+        List<String> removedRequestedScopes = new ArrayList<>();
+        String[] requestedOIDCScopes = getRequestedOIDCScopes(requestedScopes);
```

Comment:
Won't it be optimal if we get the OIDC scopes and simply filter using java streams or something?

Reusing the getRequestedOIDCScopes() does not add any value to the expected output of this method.